lib/workers/repository/process/extract-update.ts

```diff
@@ -9,16 +9,17 @@ import { PackageFile } from '../../../manager/common';
 import { RenovateConfig } from '../../../config';
 import { BranchConfig } from '../../common';
 
-export type ExtractAndUpdateResult = {
-  res: WriteUpdateResult | undefined;
+export type ExtractResult = {
   branches: BranchConfig[];
   branchList: string[];
-  packageFiles?: Record<string, PackageFile[]>;
+  packageFiles: Record<string, PackageFile[]>;
+};
+
+export type UpdateResult = {
+  res: WriteUpdateResult | undefined;
 };
```

Comment:
I think we should directly return  `WriteUpdateResult | undefined` and remove this wrapper completely

Reply:
```suggestion
```